src/Contract/Processor/NonPhpFileProcessorInterface.php

```diff
@@ -0,0 +1,20 @@
+<?php
+declare(strict_types=1);
+
+namespace Rector\Core\Contract\Processor;
+
+use Symplify\SmartFileSystem\SmartFileInfo;
+
+interface NonPhpFileProcessorInterface
+{
+    public function process(SmartFileInfo $smartFileInfo): ?string;
+
+    public function canProcess(SmartFileInfo $smartFileInfo): bool;
```

Comment:
Borrowing naming from PHP CS Fixer: https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/cbde9be1d04d46ca351e24cf39067a63326b7adf/src/Fixer/FixerInterface.php#L76

```suggestion
    public function supports(SmartFileInfo $smartFileInfo): bool;
```